### PR TITLE
SONIC-1385: fix missing addSubview when titleView & title are set

### PR DESCRIPTION
**Pull request by dennis-li-skyscanner**
# Changes
- Fix titleView & TitleLabel removeFromSuperView & addSubView logic during updateLookAndFeel
  - When the title is set, the old titleView should be removed 
  - When the titleView is set, the old titleLabel should be removed 
  - When the titleView is set, the old titleView should be removed 

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `README.md`
+ [x] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/Rating/Classes/BPKRating.swift [hunk 1]
```diff
@@ -234,6 +237,14 @@ public class BPKRating: UIView {
         }
     }
 
+    private func updateTitleSubtitleStackViewSubviews() {
+        let titleViewUpdated = titleSubtitleStackView.arrangedSubviews != [titleView, subtitleLabel]
```
Comment (frugoman) [inline Backpack/Rating/Classes/BPKRating.swift:241]: I think this boolean represents if the stack needs to be updated, rather than if it was updated. can you rename it to reflect this?

File: Backpack/Rating/Classes/BPKRating.swift [hunk 2]
```diff
@@ -208,6 +209,8 @@ public class BPKRating: UIView {
 
         updateSize()
         ratingValueLabel.text = ratingScale.displayedValue(from: value)
+
+        updateTitleSubtitleStackViewSubviews()
```
Comment (frugoman) [inline Backpack/Rating/Classes/BPKRating.swift:213]: I dont think you need to call this from here, the title and subtitle labels are added at setup time and are never removed

File: Backpack/Rating/Classes/BPKRating.swift [hunk 3]
```diff
@@ -234,6 +237,14 @@ public class BPKRating: UIView {
         }
     }
 
+    private func updateTitleSubtitleStackViewSubviews() {
+        let titleViewUpdated = titleSubtitleStackView.arrangedSubviews != [titleView, subtitleLabel]
+        if titleViewUpdated {
+            titleSubtitleStackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+            [titleView, subtitleLabel].forEach(titleSubtitleStackView.addArrangedSubview(_:))
+        }
```
Comment (frugoman) [inline Backpack/Rating/Classes/BPKRating.swift:245]: I dont quite understand why is this needed, do we really need this?

**Review by frugoman — CHANGES_REQUESTED** [review]

File: Backpack/Rating/Classes/BPKRating.swift (hunk 1 above)
Comment (dennis-li-skyscanner) [reply Backpack/Rating/Classes/BPKRating.swift]: I updated it in this commit 6a552388575a0253b740462a2c7bf4006b24fab9
